Tests/scripts/configure_tests.py

```diff
@@ -670,30 +825,33 @@ def get_test_list(files_string, branch_name):
             tests.add(test)
 
     if is_conf_json:
-        tests = tests.union(get_test_from_conf(branch_name))
-
-    if all_tests:
-        print_warning('Running all tests due to: {}'.format(','.join(all_tests)))
-        tests.add("Run all tests")
+        tests = tests.union(get_test_from_conf(branch_name, conf))
 
     if sample_tests:  # Choosing 3 random tests for infrastructure testing
         print_warning('Collecting sample tests due to: {}'.format(','.join(sample_tests)))
-        test_ids = get_test_ids(check_nightly_status=True)[0]
-        rand = random.Random(files_string + branch_name)
-        while len(tests) < 3:
-            tests.add(rand.choice(test_ids))
+        tests = tests.union(
+            get_runnable_tests(tests_num=RANDOM_TESTS_NUM, conf=conf, id_set=id_set, server_version=two_before_ga_ver))
 
     if not tests:
-        if modified_files or modified_tests_list or all_tests:
+        if modified_files or modified_tests_list:
             print_error(
                 "There is no test-playbook that checks the changes you've done, please make sure you write one.")
             global _FAILED
             _FAILED = True
+        elif changed_common:
+            print_warning('Adding 3 random tests due to: {}'.format(','.join(changed_common)))
+            tests = tests.union(get_runnable_tests(tests_num=RANDOM_TESTS_NUM, conf=conf, id_set=id_set,
+                                                   server_version=two_before_ga_ver))
         else:
             print_warning("Running Sanity check only")
+            tests = get_runnable_tests(tests_num=RANDOM_TESTS_NUM, conf=conf, id_set=id_set,
+                                       server_version=two_before_ga_ver)
             tests.add('DocumentationTest')  # test with integration configured
             tests.add('TestCommonPython')  # test with no integration configured
 
+    elif changed_common:
```

Comment:
otherwise, you will add this only when there are no tests
```suggestion
    if changed_common:
```

Reply:
As it  is this has to be elif, otherwise we'll have `TestCommonPython` twice. We can however remove 850, and do as you suggested, would that be better?